lib/widgets/app.dart

```diff
@@ -161,6 +165,40 @@ class _ZulipAppState extends State<ZulipApp> with WidgetsBindingObserver {
     super.initState();
     WidgetsBinding.instance.addObserver(this);
     UpgradeWelcomeDialog.maybeShow();
+
+    // Move to a AndroidShareService or similar.
```

Comment:
Agreed, this would be a helpful bit of code organization.

I think a good home for that would be in lib/widgets/share.dart . It's fundamentally closely tied to the SharePage widget.